web/packages/teleport/src/services/history/history.ts

```diff
@@ -53,15 +53,22 @@ const history = {
     window.location.reload();
   },
 
-  goToLogin(rememberLocation = false) {
+  goToLogin(rememberLocation = false, withAccessChangedMessage = false) {
     let url = cfg.routes.login;
+
+    // withAccessChangedMessage determines whether the login page the user is redirected to should include a notice that
+    // they were logged out due to their roles having changed.
+    if (withAccessChangedMessage) {
+      url = `${url}?accessChanged`;
+    }
+
     if (rememberLocation) {
       const { search, pathname } = _inst.location;
       const knownRoute = this.ensureKnownRoute(pathname);
       const knownRedirect = this.ensureBaseUrl(knownRoute);
       const query = search ? encodeURIComponent(search) : '';
 
-      url = `${url}?redirect_uri=${knownRedirect}${query}`;
+      url = `${url}${withAccessChangedMessage ? '&' : '?'}redirect_uri=${knownRedirect}${query}`;
```

Comment:
I wanted to say that this seems error-prone, and it looks like it really is: we miss `accessChanged` param.
Could we instead construct the params with `URLSearchParams`?
Another thing is that for `redirect_url` we use different casing than for `accessChanged`. I'd unify them.

Reply:
> it looks like it really is: we miss `accessChanged` param.

Could you clarify what you mean by this? If `withAccessChangedMessage` is true, the `url` string would already have the `accessChanged` param in it, which is why we would append the `redirect_uri` param with a `&` instead of a `?`

My initial approach was to refactor this to construct the params with `URLSearchParams` as you suggested, however the issue with that was that this would encode the redirect URL. The existing implementation has the `redirect_url` param unencoded, which seemed intentional, and I didn't want to risk breaking our existing redirect functionality.

Reply:
Ah, I see. I missed that we modify the same variable. 
If we can't use `URLSearchParams`, maybe at least we could have a simpler version of it? (note, I didn't test it)
```ts
  goToLogin(rememberLocation = false, withAccessChangedMessage = false) {
    const params: string[] = [];

    // withAccessChangedMessage determines whether the login page the user is redirected to should include a notice that
    // they were logged out due to their roles having changed.
    if (withAccessChangedMessage) {
      params.push('accessChanged');
    }

    if (rememberLocation) {
      const { search, pathname } = _inst.location;
      const knownRoute = this.ensureKnownRoute(pathname);
      const knownRedirect = this.ensureBaseUrl(knownRoute);
      const query = search ? encodeURIComponent(search) : '';
      params.push(`redirect_uri=${knownRedirect}${query}`);
    }

    const queryString = params.join('&');
    const url = queryString
      ? `${cfg.routes.login}?${queryString}`
      : cfg.routes.login;

    this._pageRefresh(url);
  },
```
By doing this, we would avoid repeating the same condition in different `if` branches.